packages/components/pie-radio-group/test/component/pie-radio-group.spec.ts

```diff
@@ -1,17 +1,151 @@
-
-        // Assert
         expect(radioGroup).toBeVisible();
     });
+
+    test.describe('Props', () => {
+        test.describe('disabled', () => {
+            test.describe('when true', () => {
+                test('should disable the slotted radio component', async ({ mount }) => {
+                    const component = await mount(PieRadioGroup, {
+                        props: {
+                            disabled: true,
+                        } as RadioGroupProps,
+                        slots: {
+                            default: '<pie-radio value="radio-one">radio 1</pie-radio>',
+                        },
+                    });
+
+                    const radio = component.locator(radioSelector);
+                    expect(radio).toBeDisabled();
+                });
+            });
+
+            test.describe('when false', () => {
+                test('should keep the slotted radio component enabled', async ({ mount }) => {
+                    const component = await mount(PieRadioGroup, {
+                        props: {
+                            disabled: false,
+                        } as RadioGroupProps,
+                        slots: {
+                            default: '<pie-radio value="radio-one">radio 1</pie-radio>',
+                        },
+                    });
+
+                    const radio = component.locator(radioSelector);
+                    expect(radio).not.toBeDisabled();
+                });
+
+                test('should disable the radio component if it has the `disabled` attribute', async ({ mount }) => {
+                    const component = await mount(PieRadioGroup, {
+                        props: {
+                            disabled: false,
+                        } as RadioGroupProps,
+                        slots: {
+                            default: '<pie-radio disabled value="radio-one">radio 1</pie-radio>',
+                        },
+                    });
+
+                    const radio = component.locator(radioSelector);
+                    expect(radio).toBeDisabled();
+                });
+            });
+        });
+
+        test.describe('value', () => {
+            test('should select the radio matching the `value` prop', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieRadioGroup, {
+                    props: {
+                        value: 'radio-two',
+                    } as RadioGroupProps,
+                    slots: {
+                        default: radioElements,
+                    },
+                });
+
+                // Act
+                const firstRadio = component.locator('[data-test-id="pie-radio-one"] input');
+                const secondRadio = component.locator('[data-test-id="pie-radio-two"] input');
+
+                // Assert
+                expect(await firstRadio.isChecked()).toBe(false);
```

Comment:
Instead of selecting the input we can probably just select `pie-radio` and check that it is `checked`?

Reply:
If that works then awesome, I remember playwright being rubbish at this particular kind of querying when its a web component in the past

Reply:
i think using `isChecked()` is slightly better here as we rely on the playwright checks for an element being checked instead of our attribute.